### DISCUSSION TITLE
Akka <-> Pekko wire compatibility

### DISCUSSION THREAD

**Discussion by levinson** [apache/pekko#586] [General]
The 1.0.1 release of pekko is not wire compatible with akka 2.6.x -- which prevents a rolling upgrade or mixed akka/pekko cluster.

The following related issues have been created:
* https://github.com/apache/incubator-pekko/issues/108 (Provide a way to handle upcoming remote wire incompatibility)
* https://github.com/apache/incubator-pekko/issues/146 (Investigate forming mixed Pekko and Akka cluster)

The proposal by @mdedetrich to configure the sendPrefix and acceptPrefix would be ideal and allow for mixed-mode clusters and migration off akka 2.6 without downtime.

I understand this feature has not been prioritized for 1.0.x release (thus far), and I'm wondering if there is any known workaround (perhaps using reflection?) or if someone could point me to the code where this could be managed by pekko-remote (both for artery and netty). Thanks!

**Comment by pjfanning** [comment]
Have a look in the `remote` src/main directory. There are only 2 places where you will find the string `"pekko"` (with the double quotes). Start with these.

Any solution that we would accept as a Pekko patch would probably require:
* that the URL schemes are configurable (reference.conf)
* that the Pekko nodes will form clusters with nodes using either `pekko:` or `akka:`
    * this allows you to start by configuring all the Pekko nodes to use `akka:` addresses so that they can fool real Akka nodes into interacting with them
    * once you have managed to replace all the real Akka nodes with Pekko nodes pretending to be Akka nodes, then you can start replacing some of these Pekko nodes with Pekko nodes that are using `pekko:` addresses - until all the nodes are changed to use `pekko:` addresses
    * we probably don't want to be forced to allow the `pekko:` scheme to be configurable in Pekko 2.0.0 so we need a way to allow users to migrate away from having Pekko nodes that pretend to be Akka nodes
    * we would need to have good test coverage with mixed nodes

Anyone is welcome to fork Pekko and modify their forks to go back to using `akka:` addresses. There is a higher bar for what we would merge into Pekko itself.

**Comment by He-Pin** [comment]
May be a simple address transformer spi, which will solve this problem too.

**Comment by pjfanning** [comment]
closing as a duplicate of https://github.com/apache/incubator-pekko/issues/108
